docs/tutorials/forum-application/3.send-message.md

```diff
@@ -0,0 +1,381 @@
+# Send Message
+
+**In this section you will learn how a user can send a message on the Forum Application.**
+
+## CheckTx
+
+In ABCI, the `CheckTx` method is used to check the validity of an individual transaction before it is included in the mempool.
+Every node runs CheckTx before letting a transaction into its local mempool.
+
+The `CheckTx` method is responsible for performing any necessary validation checks on the transaction, such as verifying
+the signature, checking for double spending, or enforcing application-specific rules. It is a lightweight and fast
+operation, as it is meant to quickly determine whether a transaction is valid or not.
+
+The method takes in a `RequestCheckTx` object, which contains the transaction to be checked (`req.Tx`) and any other
+relevant information that the application needs to validate the transaction.
+
+The `CheckTx` method should return a `ResponseCheckTx` object, which indicates whether the transaction is valid or not.
+If the transaction is valid, the response may include additional information about the transaction
+or any changes to the application state resulting from the validation.
+
+**Following is the code for the `CheckTx` method:**
+
+```go
+// CheckTx handles validation of inbound transactions. If a transaction is not a valid message, or if a user
+// does not exist in the database or if a user is banned it returns an error
+func (app ForumApp) CheckTx(_ context.Context, req *abci.CheckTxRequest) (*abci.CheckTxResponse, error) {
+    fmt.Println("Executing Application CheckTx")
```

Comment:
Can we replace `fmt.Println` by a logger (all over)?

Reply:
I didn't want to because it was a lot of work. But I did anyways, I think it looks better. But big refactoring of the app code and updating the code in the docs is cumbersome. In the future need to think a better way to handle these. But I replaced everything with log messages. 😅